pkg/controller/external.go

```diff
@@ -155,13 +159,19 @@ func (e *external) Observe(ctx context.Context, mg xpresource.Managed) (managed.
 		return managed.ExternalObservation{}, errors.New(errUnexpectedObject)
 	}
 
+	policySet := sets.New[xpv1.ManagementAction](tr.GetManagementPolicies()...)
+
 	// Note(turkenh): We don't need to check if the management policies are
 	// enabled or not because the crossplane-runtime's managed reconciler already
-	// does that for us. In other words, if the management policy is set to
-	// "ObserveOnly" without management policies being enabled, the managed
+	// does that for us. In other words, if the management policies are set
+	// without management policies being enabled, the managed
 	// reconciler will error out before reaching this point.
 	// https://github.com/crossplane/crossplane-runtime/pull/384/files#diff-97300a2543f95f5a2ada3560bf47dd7334e237e27976574d15d1cddef2e66c01R696
-	if tr.GetManagementPolicy() == xpv1.ManagementObserveOnly {
+	// Note (lsviben) We are only using import instead of refresh if the
+	// management policies do not contain create or update as they need the
+	// required fields to be set, which is not the case for import.
+	if tr.GetManagementPolicies() != nil &&
+		!policySet.HasAny(xpv1.ManagementActionCreate, xpv1.ManagementActionUpdate, xpv1.ManagementActionAll) {
```

Comment:
Wondering if we could use the newly introduced `ManagementPoliciesResolver` in runtime here?

Reply:
Well, we would need to add the enabled flag and deletion policy into it. For these simple cases it seemed to me like its not needed